src/main/java/com/powsybl/openloadflow/ac/DistributedLoadSlackOuterLoop.java

```diff
@@ -0,0 +1,149 @@
+                .map(bus -> new ParticipatingLoad(bus, bus.getParticipationFactor()))
+                .collect(Collectors.toList());
+    }
+
+    private void normalizeLoadParticipationFactors(List<ParticipatingLoad> participatingLoads) {
+        double factorSum = participatingLoads.stream()
+                .mapToDouble(participatingLoad -> participatingLoad.factor)
+                .sum();
+        if (factorSum == 0) {
+            throw new PowsyblException("No more load participating to slack distribution");
+        }
+        for (ParticipatingLoad participatingLoad : participatingLoads) {
+            participatingLoad.factor /= factorSum;
+        }
+    }
+
+    @Override
+    public OuterLoopStatus check(OuterLoopContext context) {
+        double slackBusActivePowerMismatch = context.getLastNewtonRaphsonResult().getSlackBusActivePowerMismatch();
+        if (Math.abs(slackBusActivePowerMismatch) > DefaultNewtonRaphsonStoppingCriteria.CONV_EPS_PER_EQ) {
+            LfNetwork network = context.getNetwork();
+
+            List<ParticipatingLoad> participatingLoads = getParticipatingLoads(network);
+
+            int iteration = 0;
+            double remainingMismatch = slackBusActivePowerMismatch;
+            while (!participatingLoads.isEmpty()
+                    && Math.abs(remainingMismatch) > SLACK_P_RESIDUE_EPS) {
+
+                remainingMismatch -= run(participatingLoads, iteration, remainingMismatch);
+
+                iteration++;
+            }
+
+            if (Math.abs(remainingMismatch) > SLACK_P_RESIDUE_EPS) {
+                throw new PowsyblException("Failed to distribute slack bus active power mismatch, "
+                        + remainingMismatch * PerUnit.SB + " MW remains");
+            } else {
+                LOGGER.info("Slack bus active power ({} MW) distributed in {} iterations",
+                        slackBusActivePowerMismatch * PerUnit.SB, iteration);
+            }
+
+            return OuterLoopStatus.UNSTABLE;
+        }
+
+        LOGGER.debug("Already balanced");
+
+        return OuterLoopStatus.STABLE;
+    }
+
+    private double run(List<ParticipatingLoad> participatingLoads, int iteration, double remainingMismatch) {
+        // normalize participation factors at each iteration start as some
+        // loads might have reach zero and have been discarded.
+        normalizeLoadParticipationFactors(participatingLoads);
+
+        double done = 0d;
+        int modifiedBuses = 0;
+        int loadsAtMin = 0;
+        Iterator<ParticipatingLoad> it = participatingLoads.iterator();
+        while (it.hasNext()) {
+            ParticipatingLoad participatingLoad = it.next();
+            LfBus load = participatingLoad.load;
+            double factor = participatingLoad.factor;
+
+            double targetP = load.getLoadTargetP();
+
+            double newTargetP = targetP - remainingMismatch * factor;
```

Comment:
Be careful of the sign here I don't remenber the convention we are using.

Reply:
In fact, for a given mismatch, an increase in the production is equivalent to a decrease of loads active power and a decrease in the production is equivalent to an increase of loads. For sure, I am really not sure that my implementation makes it clear… what do you suggest ?